src/models/PolicyList.ts

```diff
@@ -134,19 +142,23 @@ class PolicyList extends EventEmitter {
     /**
      * Return all the active rules of a given kind.
      * @param kind e.g. RULE_SERVER (m.policy.rule.server). Rule types are always normalised when they are interned into the PolicyList.
+     * @param recommendation A specific recommendation to filter for e.g. `m.ban`. Please remember recommendation varients are normalized.
      * @returns The active ListRules for the ban list of that kind.
+     * FIXME: it would be nice if this was a generic to the ListRule type, (and get the recommendation type from it) but that seems hard to do, maybe impossible?.
+     * TFW you mess up type erasure so bad comeon.
      */
-    private rulesOfKind(kind: string): ListRule[] {
+    public rulesOfKind(kind: string, recommendation?: Recommendation): ListRule[] {
         const rules: ListRule[] = []
         const stateKeyMap = this.state.get(kind);
         if (stateKeyMap) {
             for (const event of stateKeyMap.values()) {
-                const rule = event?.unsigned?.rule;
-                // README! If you are refactoring this and/or introducing a mechanism to return the list of rules,
-                // please make sure that you *only* return rules with `m.ban` or create a different method
-                // (we don't want to accidentally ban entities).
-                if (rule && rule.kind === kind && rule.recommendation === Recommendation.Ban) {
-                    rules.push(rule);
+                const rule = event[PolicyList.EVENT_RULE_ANNOTATION_KEY];
+                if (rule && rule.kind === kind) {
+                    if (recommendation === undefined) {
```

Comment:
Why do we do that? Looks like this changes the existing behavior.

Reply:
We do this so you can get all the policy types for one kind without needing to specify a recommendation, which is useful for `allRules`, `allUserRules` etc... All of the existing uses of these were checked too.
The use case for them is querying the list with `rules matching` and the rules dump command itself.

Reply:
If we didn't do this, we'd have to refactor the `rules` and `rules matching` command